btc/path_wallet.go

```diff
@@ -0,0 +1,137 @@
+package btc
+
+import(
+	"context"
+	"errors"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	"github.com/btcsuite/btcutil/hdkeychain"
+)
+
+type wallet struct {
+	Network string
+	MasterKey string
+	DerivationPath []uint32
+}
+
+func pathWallet(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "wallet/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"network": &framework.FieldSchema{
+				Type: framework.TypeString,
+				Description: "btc network type: mainnet | testnet",
+			},
+			"name": &framework.FieldSchema{
+				Type: framework.TypeString,
+				Description: "wallet name",
+			},
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathWalletRead,
+			logical.UpdateOperation: b.pathWalletWrite,
+		},
+
+		HelpSynopsis:    pathWalletsHelpSyn,
+		HelpDescription: pathWalletsHelpDesc,
+	}
+}
+
+func (b *backend) pathWalletWrite(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	network := d.Get("network").(string)
+	if network == "" {
+		return nil, errors.New("missing network")
+	}
+
+	walletName := d.Get("name").(string)
+	if walletName == "" {
+		return nil, errors.New("missing wallet name")
+	}
+
+	// return error if a wallet with same name has already been created
+	w, err := b.GetWallet(ctx, req.Storage, walletName)
+	if err != nil {
+		return nil, err
+	}
+	if w != nil {
+		return logical.ErrorResponse("Wallet with name '" + walletName + "' already exists"), nil
+	}
+
+	wallet, err := createWallet(network)
+	if err != nil {
+		return nil, err
+	}
+	
+	// create storage entry
+	entry, err := logical.StorageEntryJSON("wallet/" + walletName, wallet)
+	if err != nil {
+		return nil, err
+	}
+
+	// save in local storage
+	if err := req.Storage.Put(ctx, entry); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (b *backend) pathWalletRead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	walletName := d.Get("name").(string)
+
+	w, err := b.GetWallet(ctx, req.Storage, walletName)
+	if err != nil {
+		return nil, err
+	}
+	if w == nil {
+		return nil, nil
+	}
+
+	key, err := hdkeychain.NewKeyFromString(w.MasterKey)
+	if err != nil {
+		return nil, err
+	}
+
+	xprv, err := derivePrivKey(key, w.DerivationPath)
+	if err != nil {
```

Comment:
Why Private Key is returned by default? Should only be possible to sweep or exported encrypted (with the root token?)

Reply:
absolutely right. Left this here for testing purpose. I'm changing it so it returns only the priv key in WIF

Reply:
In the final design, It should be a separated endpoint and only encrypted with root token or something else.

Reply:
Mainly for testing purpose. I have to change it so it returns only the key in WIF, and also add auth check